### PR TITLE
Hide product attribute selector when adding the first local attribute

### LINKED ISSUE #36665: Variable product: Remove product attribute selector
After merging [this PR](https://github.com/woocommerce/woocommerce/pull/36606), we will show the attribute creation form when there are no attributes.

screenshot-user-images githubusercontent com-2023 01 28-10_45_50

**Acceptance criteria:**

- [ ] Remove [the product attribute selector](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/admin/meta-boxes/views/html-product-data-attributes.php#L39-L58) when there are no attributes.
- [ ] Once the user adds the first attribute using the fields, they will see the regular interface with the selector, the button, and the attribute list.

### PR DESCRIPTION
**Pull request by nathanss**
### All Submissions:

-   [ ] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [ ] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

Hide product attribute selector when adding the first local attribute.

I decided to use jQuery to show the selector when the first value blurs to avoid making the user click on "Save attributes" twice. 

https://user-images.githubusercontent.com/13437655/221646376-6cab865b-862b-45d9-97a7-c179dbf18379.mov

Closes #36665 .

### How to test the changes in this Pull Request:

- [ ] Have you followed the [Writing high-quality testing instructions guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions)?

1. Have no global attributes at your Woo installation (the ones created at Products > Attributes)
2. Go to Products > Add
3. Go to the Attributes tab
4. Check if the message appears and the product selector is hidden
5. Fill the first attribute
6. Check if the selector appears

### Other information:

-   [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [ ] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?
-   [ ] Have you included testing instructions?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Review by louwie17 — CHANGES_REQUESTED** [review]
Overall this tested well, although I did notice one issue when a user clicks `Remove` right away, as the dropdown will not show up then and the user will not be able to add attributes unless they refresh:
Screenshot 2023-02-28 at 2 54 32 PM

I am also kinda questioning this change in general, to me it feels a little strange that the `dropdown` only shows up after a user enters some values and then is displayed permanently (see video in PR description).
I think we are much better of keeping it displayed permanently (as it is now) without adding extra logic to handle all these edge cases. The fact we are having it showing up somewhat randomly from a users perspective, might cause them to be more confused then helping them be less so.
cc: @octaedro @jarekmorawski

**Comment by nathanss** [comment]
@louwie17 I am also not so confident about this change, especially since the current architecture makes it very hard to maintain consistency, it seems very easy to let bugs and edge cases like this slip.

**Comment by mattsherman** [comment]
> although I did notice one issue when a user clicks Remove right away, as the dropdown will not show up then and the user will not be able to add attributes unless they refresh

This is definitely a case that should be addressed if we proceed with this change.

I'm on the fence as to whether this change should be made at this point in time. On the one hand, it does (potentially) slightly reduce confusion in the case where there are no global attributes and no custom attributes.

But, the overall UX of this section is still confusing. You can still click "Add" multiple times after the first attribute is filled out, and get multiple empty attribute fields... when saving these are ignored though, so it isn't a big deal.

**Review by mattsherman — CHANGES_REQUESTED** [review]
The scenario where the user clicks "Remove" when there is only one attribute and the dropdown doesn't appear needs to be addressed.

Other than that, I think this change works as expected.

File: plugins/woocommerce/client/legacy/js/admin/meta-boxes-product.js [hunk 1]
```diff
@@ -467,11 +471,18 @@ jQuery( function ( $ ) {
...
 
-			$attributes
+			const lastAttribute = $attributes
 				.find( '.woocommerce_attribute' )
-				.last()
-				.find( 'h3' )
-				.trigger( 'click' );
+				.last();
+
+			lastAttribute.find( 'h3' ).trigger( 'click' );
```
Comment (mattsherman) [inline plugins/woocommerce/client/legacy/js/admin/meta-boxes-product.js:478]: Why is this needed?

File: plugins/woocommerce/client/legacy/js/admin/meta-boxes-product.js [hunk 2]
```diff
@@ -467,11 +471,18 @@ jQuery( function ( $ ) {
...
+
+			if ( $attributes.length === 1 ) {
+				// if there is only one attribute, show the actions that were hidden when user finishes typing attribute value to avoid a roundtrip
+				lastAttribute.find( '[name^="attribute_values"]' ).on( 'blur', function () {
+					$( '#product_attributes_actions' ).removeClass( 'hidden' );
+				} );
+			}
```
Comment (mattsherman) [inline plugins/woocommerce/client/legacy/js/admin/meta-boxes-product.js:485]: Should we do this when `$attributes.length > 0` to be a bit more robust and handle some unexpected case somehow more than one attribute was added?

File: plugins/woocommerce/client/legacy/js/admin/meta-boxes-product.js [hunk 3]
```diff
@@ -57,6 +57,10 @@ jQuery( function ( $ ) {
 		} );
 	} );
 
+	$( document.body ).on( 'woocommerce_attributes_saved', function () {
+		$( '#product_attributes_actions' ).removeClass( 'hidden' );
```
Comment (mattsherman) [inline plugins/woocommerce/client/legacy/js/admin/meta-boxes-product.js:61]: The UI allows the attributes to be saved when not filled out. So, a save can result in zero attributes. We should check the number of attributes before showing these actions.

**Comment by jarekmorawski** [comment]
> I am also kinda questioning this change in general, to me it feels a little strange that the dropdown only shows up after a user enters some values and then is displayed permanently (see video in PR description).

The idea was that the user would only see the form when they first enter this screen. They wouldn't see Remove button or the dropdown. The dropdown doesn't make much sense because what there's nothing we could show there. Can we hide them?

**Comment by nathanss** [comment]
@jarekmorawski 

The objection that I have to this change is that it would make the user have to click twice in 'Save attributes' to add more than one Attribute if I didn't take the liberty of showing the toolbar again after the first attribute value is blurred:

https://user-images.githubusercontent.com/13437655/222134955-6a5e0f46-6acb-4292-abfa-117350a84d45.mov

It seems that this would make the experience worse for users that want to add more than one attribute.

On a related note, a lot of improvements that would be very simple with React can easily get complicated with the current architecture: Check this internal P2: pcQJnU-1if-p2

**Comment by jarekmorawski** [comment]
I see your point. Clicking the save button twice adds unnecessary friction, and I'm sure there's a way to improve the whole flow without over-complicating the experience.

I'd like to spend a bit more time thinking about this. @pmcpinto, would you be okay with pausing this work for a while so we can explore a bit and see if there're other improvements we can make?

**Comment by pmcpinto** [comment]
> I'd like to spend a bit more time thinking about this. @pmcpinto, would you be okay with pausing this work for a while so we can explore a bit and see if there're other improvements we can make?

Yes, let's do it

**Comment by jarekmorawski** [comment]
I added extra flows and designs to https://github.com/woocommerce/woocommerce/issues/37099.

**Comment by octaedro** [comment]
[This epic](https://github.com/woocommerce/woocommerce/issues/37099) changes a little bit the flow tackled by this PR so I'm going to close it.

**Pull request closed without merging**